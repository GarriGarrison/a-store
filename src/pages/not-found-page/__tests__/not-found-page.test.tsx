import { render, screen } from '@testing-library/react'
import {BrowserRouter as Router} from 'react-router-dom'
import { NotFoundPage } from '..'


describe('Тесты страницы «404»', () => {
  
  test('рендер заголовка H1', () => {
    render(
      <Router>
        <NotFoundPage />
      </Router>
    )
    
    const element = screen.getByText('Страница не найдена')
    expect(element).toBeInTheDocument()
  })

  test('отображение весёлой картинки', () => {
    render(
      <Router>
        <NotFoundPage />
      </Router>
    )

    const testImage = screen.getByRole('img') as HTMLImageElement
    expect(testImage.alt).toMatch(/not found/i)
  })

  test('снепшот страницы', () => {
    const { baseElement } = render(
      <Router>
        <NotFoundPage />
      </Router>
    )
    
    expect(baseElement).toMatchSnapshot()
  })
})
