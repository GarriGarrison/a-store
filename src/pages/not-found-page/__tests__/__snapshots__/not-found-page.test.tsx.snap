// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты страницы «404» снепшот страницы 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <h1
        class="typography__component_lu6tt typography__styrene-xlarge_ez5iu typography__medium_lu6tt"
      >
        Страница не найдена
      </h1>
      <img
        alt="Not Found"
        src="notFound.png"
      />
    </div>
  </div>
</body>
`;
